test/Test/Pos/Block/Logic/VarSpec.hs

```diff
@@ -283,3 +296,63 @@ verifyBlockScenarioResult = \case
             "Block scenario resulted in a change to the blockchain" <>
             " relative to the " <> show snapId <> " snapshot:\n" <>
             show dbDiff
+
+----------------------------------------------------------------------------
+-- Forks
+----------------------------------------------------------------------------
+
+singleForkSpec :: ForkDepth -> Spec
+singleForkSpec fd = do
+    prop singleForkDesc (singleForkProp fd)
+  where
+    singleForkDesc =
+      "a blockchain of length q=0..(9.5*k) blocks can switch to a fork " <>
+      "of length j>i with a common prefix i, d=q-i"
+
+singleForkProp :: ForkDepth -> BlockProperty ()
+singleForkProp fd = do
+    scenario <- blockPropertyScenarioGen $ genSingleFork fd
+    runBlockScenarioAndVerify scenario
+
+data ForkDepth = ForkShort | ForkMedium | ForkDeep
+
+genSingleFork :: forall g m. (RandomGen g, Monad m) => ForkDepth -> BlockEventGenT g m ()
+genSingleFork fd = do
+    let k = fromIntegral blkSecurityParam
+    d <- getRandomR $ case fd of
+        ForkShort  -> (1, k - 1)
+        ForkMedium -> (max 1 (k-2), k+2)
+        ForkDeep   -> (k+1, div (k*3) 2)
```

Comment:
In `ForkDeep` and `ForkShort` case the first value will be greater than the second one if `k = `, but perhaps we don't care.